### PR TITLE
Feat/component detail analytics

### PR DESCRIPTION
**Pull request by roshaans**
Changes to Component Details Page. 

Things to do 

- [x] Use QueryAPI graphQL to pull developer information 

Head to near.org/<yourNearAccount>/widget/ComponentDetailsPage?src=devgovgigs.near/widget/gigs-board.pages.Feed to test

image

### CODE REVIEW

**Comment by calebjacob** [comment]
@roshaans Is there an associated ticket for this work?

File: src/ComponentDetailsPage.jsx [hunk 1]
```diff
@@ -19,6 +23,36 @@ const metadata = data.metadata;
...
+    numberOfForks: 10,
+    impressions: 93,
+    parents: [
+      "mob.near/widget/Image",
+      "mob.near/widget/Image",
+      "mob.near/widget/Image",
+      "mob.near/widget/Image",
+    ],
+  },
+};
```
Comment (calebjacob) [inline src/ComponentDetailsPage.jsx:55]: Why is this data hardcoded?

File: src/ComponentDetailsPage.jsx [hunk 2]
```diff
@@ -176,166 +381,176 @@ if (!exists) {
...
-
-          {tags.length > 0 && (
-            <div>
-              <SmallTitle>Tags</SmallTitle>
+          </Content>
+        )}
+      </Wrapper>
+      <Sidebar>
+        <SideBarContainer>
+          <SmallTitle style={{ "padding-top": "20px" }}>DEVELOPER</SmallTitle>
```
Comment (calebjacob) [inline src/ComponentDetailsPage.jsx:463]: Any time we want to render a string in all uppercase, we should rely on CSS to do the transformation with `text-transform: uppercase` and write the text as `Developer`. Writing the text in all uppercase can cause accessibility issues for screen readers.

**Review by calebjacob — CHANGES_REQUESTED** [review]

File: src/ComponentDetailsPage.jsx (hunk 1 above)
Comment (roshaans) [reply src/ComponentDetailsPage.jsx:55]: Should have commented on this review before. 

This is a final API endpoint we are waiting on, but I wanted to get a general review of this PR before that.

File: src/ComponentDetailsPage.jsx [hunk 3]
```diff
@@ -176,166 +381,176 @@ if (!exists) {
...
               <Widget
-                src="${REPL_ACCOUNT}/widget/Tags"
-                props={{
-                  tags,
-                }}
+                src="${REPL_ACCOUNT}/widget/SocialMarkdown"
+                props={{ text: accountProfileDescription }}
               />
-            </div>
+            </Bio>
```
[the remark this answers is not in the document]
Comment (roshaans) [reply src/ComponentDetailsPage.jsx:477]: We only show a maximum of 25 words so it should not overflow

https://github.com/near/near-discovery-components/pull/408/files#diff-9133aecb390e324a27b0ed7dceb59296ddab70338a02a57fc4109896bd8d70eaR32

**Comment by roshaans** [comment]
> @roshaans Is there an associated ticket for this work?

yes, [Here it is ](https://github.com/near/data-platform/issues/23)

File: src/ComponentDetailsPage.jsx (hunk 1 above)
Comment (calebjacob) [reply src/ComponentDetailsPage.jsx:54]: Sounds good! Maybe in the future, you could mark your PR as a draft if there's still pieces you need to finalize.

**Comment by calebjacob** [comment]
Thanks for linking the ticket @roshaans - I guess I'm curious what the product reasoning is behind renaming and removing some of the tabs. I didn't see any discussion about that in the linked ticket.

File: src/ComponentDetailsPage.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (calebjacob) [reply src/ComponentDetailsPage.jsx:476]: That definitely helps. I guess my main concern are descriptions that contain markdown or tags like `<h1>`.

**Comment by roshaans** [comment]
> Thanks for linking the ticket @roshaans - I guess I'm curious what the product reasoning is behind renaming and removing some of the tabs. I didn't see any discussion about that in the linked ticket.

@TiffanyGYJ can provide more insights here

**Comment by TiffanyGYJ** [comment]
> Thanks for linking the ticket @roshaans - I guess I'm curious what the product reasoning is behind renaming and removing some of the tabs. I didn't see any discussion about that in the linked ticket.

@calebjacob We are doing a special NEARCON App Upvoting Event, more details here 
[Google Doc on the event ](https://docs.google.com/document/d/16uQdR0iRcWq8kyBFgUauATdWq7HMPQYMiNZyPnPkG9o/edit?usp=sharing)
[Blog Post with simplified explanation for DApp owners](https://docs.google.com/document/d/1-T1npHHg93ZI6JXZ-z2Z8t4Mm9xV_sLaOE4an_6lLKw/edit?usp=sharing ) 

TLDR: 
- this is a NEARCON Event in the form of a NCON Task that ask people to upvote qualified apps and decide who gets featured 
- We are using a fork of the App Library and populate it with Apps that are qualified (description required + tags<Earn,play,develop,engage> required, so we are running a data filtering pipeline on the back to grab those. 
- Then we will enable a upvote button on the details page of these qualified apps.
- Users at NEARCon can visit this App Upvoting Event component, look at the qualified apps, upvote the ones they like.
- Result is top 8 upvotes will get featured in the App Library (near.org/applications)

**Comment by calebjacob** [comment]
Thanks for the info @TiffanyGYJ. Going forward, it would be nice to have PR's link to issues with those requirements listed so that other developers can understand the context for the changes.

**Comment by TiffanyGYJ** [comment]
> Thanks for the info @TiffanyGYJ. Going forward, it would be nice to have PR's link to issues with those requirements listed so that other developers can understand the context for the changes.

Yep Great advice, will include these in the future, thank you for the help @calebjacob !

File: src/TimeAgoComponentDetailsPage.jsx
[the remark this answers is not in the document]
Comment (roshaans) [reply src/TimeAgoComponentDetailsPage.jsx:1]: The TimeAgo component we currently have is mostly used in `posts` and shows time ranges older than a day as a date. 

For this component we would like to use another strategy which is to show all time ranges instead of showing dates. 

examples

```

Assuming a block timestamp corresponding to 7 months ago:
- orignal timeago component: Would show the specific date, e.g., "Mar 15".
- new timeago component: "7 months"
----
Assuming a block timestamp corresponding to 2 years ago:
- orignal timeago component:: Would show the specific date with year, e.g., "Oct 24, 2021".
- new timeago component: "2 yrs"
```

[the remark this answers is not in the document]
Comment (roshaans) [reply src/TimeAgoComponentDetailsPage.jsx:1]: I did not have time to merge the changes and figure out weather this change in the original time ago component would have uninteded side effects which is why I created a new component. But I totally agree with you. we should use a prop to differentiate this behavior.

[the remark this answers is not in the document]
Comment (calebjacob) [reply src/TimeAgoComponentDetailsPage.jsx:1]: No problem @roshaans - I pushed a commit to merge the two versions into a single `TimeAgo` component. You can use the `alwaysRelative: true` prop use your formatting for other use cases.

**Review by calebjacob — APPROVED** [review]
Looks good!
